packages/server/src/plugin/subscriptionCallback/index.ts

```diff
@@ -0,0 +1,655 @@
+        // will be done in `willSendResponse`. The router expects the initial
+        // response to be a 200, with the `subscription-protocol` header set to
+        // `callback`.
+        async responseForOperation({ request }) {
+          logger?.debug(
+            'Received new subscription request',
+            request.extensions!.subscription.subscription_id,
+          );
+
+          return {
+            http: {
+              status: 200,
+              headers: new HeaderMap([['subscription-protocol', 'callback']]),
+            },
+            body: {
+              kind: 'single',
+              singleResult: {},
+            },
+          };
+        },
+        // The majority of the subscription work is implemented in this hook.
+        // This can _almost_ all be implemented in `responseForOperation`,
+        // however in the case of GraphQL validation errors,
+        // `responseForOperation` won't actually be called, but
+        // `willSendResponse` will.
+        async willSendResponse({
+          request,
+          schema,
+          document,
+          contextValue,
+          operationName,
+          response,
+        }) {
+          const {
+            callback_url: callbackUrl,
+            subscription_id: id,
+            verifier,
+          } = request.extensions!.subscription;
+          try {
+            logger?.debug(
+              'Sending `check` request to router',
+              request.extensions!.subscription.subscription_id,
+            );
+            // Before responding to the original request, we need to complete a
+            // roundtrip `check` request to the router, so we `await` this
+            // request.
+            await subscriptionManager.checkRequest({
+              callbackUrl,
+              id,
+              verifier,
+            });
+            logger?.debug(
+              '`check` request successful',
+              request.extensions!.subscription.subscription_id,
+            );
+          } catch (e) {
+            const graphqlError = ensureGraphQLError(e);
+            logger?.error(
+              `\`check\` request failed: ${graphqlError.message}`,
+              id,
+            );
+            // In the event of a check failure, we respond to the original
+            // request with a >=400 status code.
+            if (response.body.kind === 'single') {
+              response.body.singleResult.errors = [graphqlError];
+              response.http.status = 400;
+            }
+            return;
+          }
+
+          // The `check` request was successful, so we can initialize the actual
+          // `graphql-js` subscription. We don't expect `subscribe` to throw,
+          // but rather return an object with `errors` on it (if there are any).
+          logger?.debug(`Starting graphql-js subscription`, id);
+          let subscription: Awaited<ReturnType<typeof subscribe>>;
+          try {
+            subscription = await subscribe({
```

Comment:
I think this execution won't end up invoking any other plugin-y things (like, willResolveField especially). Is this intentional? Will users expect this?

Relatedly, does our subscription protocol track field executions (via something like ftv1) at all, or is that not part of the system at this time?